File: 2023-11-15-multiple-sns-login-with-spring-security-oauth2-client/action-in-blog/src/main/java/action/in/blog/controller/HomeController.java
package action.in.blog.controller;

import action.in.blog.domain.token.AuthenticatedUser;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
public class HomeController {

    @GetMapping("/home")
    public ModelAndView home(@AuthenticationPrincipal AuthenticatedUser user) {
        var mav = new ModelAndView("home");
        mav.addObject("email", user.getEmail());
        mav.addObject("nickName", user.getNickName());
        return mav;
    }
}

File: 2025-03-25-aws-alb-application-protocol-change-and-spring-base-url-problem/action-in-blog/src/main/kotlin/action/in/blog/controller/HomeController.kt
package action.`in`.blog.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HomeController {

    @GetMapping("/health")
    fun index() = "ok"

    @GetMapping("/home")
    fun home() = "Hello World"
}